cmd/node-termination-handler.go

```diff
@@ -150,14 +151,15 @@ func main() {
 			time.Sleep(8 * time.Second)
 			stopCh <- struct{}{}
 		}()
+
 		//will retry 4 times with an interval of 2 seconds.
-		err = wait.PollImmediateUntil(2*time.Second, func() (done bool, err error) {
+		err = wait.PollUntilContextCancel(wait.ContextForChannel(stopCh), 2*time.Second, true, func(context.Context) (done bool, err error) {
```

Comment:
`wait.ContextForChannel()` cancels the context when `stopCh` is closed, not when it receives. An easier approach could be to drop the `stopCh` stuff and do

```suggestion
		pollCtx, cancelPollCtx := context.WithTimeout(context.Background(), 8 * time.Second)
		err = wait.PollUntilContextCancel(pollCtx, 2*time.Second, true, func(context.Context) (done bool, err error) {
```